### ISSUE TITLE
Can't complete setup - login issue

### ISSUE THREAD

**Issue by nickrout** [ggravlingen/ha-addon-scrypted#9]
Running supervised HA install on amd64.

I loaded the addon and started it.

A dialog asks me to put in a username, password and password confirmation.

After I did this a similar dialog with username and password popped up,  I entered the details from stage 1 and it just keeps repeating the second dialog, as if the login is not being accepted for some reason. 

I looked at the docker and it is clear that /dev, /config and /data are mapped into the container. Nothing is created in /data.

In /config there is a dir /config/data_scrypted but I cannot make sense of any of that!

If I stop the addon, delete /config/data_scrypted I can start again with the first dialog (user/pw/confirm pw) and then the whole business starts again.

**Comment by ggravlingen** [comment]
Odd, I had that issue myself but it was eventually solved. Do you get any errors if you open developer tools in your browser, and then click network and reload the page? An error is indicated by text in red.

**Comment by newdaddy46052** [comment]
I too am having the same issues. I just saw this when I learned about scrypted last night and figured I’d install this on my home assistant blue. I’m not too tech savvy but what do you need from me to help resolve this issue.

**Comment by ggravlingen** [comment]
Same as above please. Open dev tools in chrome, click network and do a page refresh. Any red lines?

**Comment by newdaddy46052** [comment]
> Same as above please. Open dev tools in chrome, click network and do a page refresh. Any red lines?

So when i log into my homeassistant via chrome and do as you say i do see one batch of red.
ive attached a screenshot of what i think you are requesting. Hope this helps and let me know if you need anything more.
![Scrypted_network_Red](https://user-images.githubusercontent.com/105159993/167300657-34fe27b6-6e5f-41bb-98a3-df72b8934bc3.PNG)

**Comment by ggravlingen** [comment]
> > Same as above please. Open dev tools in chrome, click network and do a page refresh. Any red lines?
> 
> So when i log into my homeassistant via chrome and do as you say i do see one batch of red. ive attached a screenshot of what i think you are requesting. Hope this helps and let me know if you need anything more. ![Scrypted_network_Red](https://user-images.githubusercontent.com/105159993/167300657-34fe27b6-6e5f-41bb-98a3-df72b8934bc3.PNG)

What's the request URL of the red line? Please remove the first part of the URL, it's the address to your server. 
`Request URL: https://[redacted].ui.nabu.casa/api/hassio_ingress/IaxnV2Jt2Zci8KAUAwGgS3_ZBNgUvD5ZkWXQV2gMECI/endpoint/@scrypted/core/public/`

**Comment by newdaddy46052** [comment]
> > > Same as above please. Open dev tools in chrome, click network and do a page refresh. Any red lines?
> > 
> > 
> > So when i log into my homeassistant via chrome and do as you say i do see one batch of red. ive attached a screenshot of what i think you are requesting. Hope this helps and let me know if you need anything more. ![Scrypted_network_Red](https://user-images.githubusercontent.com/105159993/167300657-34fe27b6-6e5f-41bb-98a3-df72b8934bc3.PNG)
> 
> What's the request URL of the red line? Please remove the first part of the URL, it's the address to your server. `Request URL: https://[redacted].ui.nabu.casa/api/hassio_ingress/IaxnV2Jt2Zci8KAUAwGgS3_ZBNgUvD5ZkWXQV2gMECI/endpoint/@scrypted/core/public/`

Referer: http://{redacted}/api/hassio_ingress/qp0z6vjaSYJamHFtiDWJy1T6WsQ56imlKhVqnQxBK_Y/endpoint/@scrypted/core/public/

is this it?

**Comment by ggravlingen** [comment]
Yes, thanks. Are there any new red lines when you type in the login/password and press login?

**Comment by newdaddy46052** [comment]
> Yes, thanks. Are there any new red lines when you type in the login/password and press login?

ok so when i try to login to scrypted it ends up getting rid of that redline and refreshes the scrypted login in page again asking to login again and just keeps doing that but the redline has since gone away when i try to re login.  ive tried changing the password in hope there may just be a simple bug but no luck. just goes back to the log in screen.

**Comment by ggravlingen** [comment]
> > Yes, thanks. Are there any new red lines when you type in the login/password and press login?
> 
> 
> 
> No additional red lines when doing as you say. Just the original 

Hm, I'm not sure what's going on here. Maybe delete the scrypted folder in /config and try again using something basic for login like username: test and password: test.

**Comment by newdaddy46052** [comment]
> > > Yes, thanks. Are there any new red lines when you type in the login/password and press login?
> > 
> > 
> > No additional red lines when doing as you say. Just the original
> 
> Hm, I'm not sure what's going on here. Maybe delete the scrypted folder in /config and try again using something basic for login like username: test and password: test.

ok it doesnt seem like i can do that so easily on my end with my homeassistant blue device as it doesnt allow me to delete folders have to delete everything in them manually so gonna work on that and ill get back to you later to see if a "fresh install" will work better.

**Comment by ggravlingen** [comment]
Have you tried the vs-code addon? It might be helpful when deleting a full folder:
https://github.com/hassio-addons/addon-vscode/blob/main/README.md

**Comment by newdaddy46052** [comment]
> Have you tried the vs-code addon? It might be helpful when deleting a full folder: https://github.com/hassio-addons/addon-vscode/blob/main/README.md

you're awesome! that made deleting much much easier! im new to all this so very much appreciated. 

after a "fresh install" im still getting the same of just refreshing of the login screen.
would logs from the scrypted addon in ha help you any?

**Comment by ggravlingen** [comment]
Yes, please logs are useful. Hit restart and then copy/paste the logs here. When that is done, go to the page of the service and see what happens in the logs. You will see a lot of test so wait until everything has been loaded and paste the last bits. Please also try to do a login and paste whatever logs show up after that.

Also, just to be sure: you are running version 1.0.5 of the addon?
![image](https://user-images.githubusercontent.com/21142447/167308095-3af84e1d-3888-4c10-b76e-19c5c7d81b25.png)

**Comment by newdaddy46052** [comment]
> Yes, please logs are useful. Hit restart and then copy/paste the logs here. When that is done, go to the page of the service and see what happens in the logs. You will see a lot of test so wait until everything has been loaded and paste the last bits. Please also try to do a login and paste whatever logs show up after that.
> 
> Also, just to be sure: you are running version 1.0.5 of the addon?

yes im running that version. ive  restarted and attached pic showing my version and notepad with the logs.
Also thank you for helping with this!
![Scryptedversion](https://user-images.githubusercontent.com/105159993/167308745-42c0ff46-6b2f-4e84-8e48-ae7d29dcf264.PNG)
[ScryptedLogs.txt](https://github.com/ggravlingen/ha-addon-scrypted/files/8647491/ScryptedLogs.txt)

**Comment by ggravlingen** [comment]
I'm getting the same output here:

```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] nginx.sh: executing... 
[cont-init.d] nginx.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
[19:53:12] INFO: Starting Scrypted...
i Arlo Camera Plugin loading @scrypted/arlo
i Arlo Camera Plugin pid 325
i Scrypted Core loading @scrypted/core
i Scrypted Core pid 326
i Rebroadcast Plugin loading @scrypted/prebuffer-mixin
i Rebroadcast Plugin pid 332
i Snapshot Plugin loading @scrypted/snapshot
i Snapshot Plugin pid 340
i WebRTC Plugin loading @scrypted/webrtc
i WebRTC Plugin pid 347
#######################################################
Scrypted Server (Local)   : https://localhost:10443/
Scrypted Server (Remote)  : https://172.30.33.3:10443/
Version:       : 0.0.176
#######################################################
Scrypted insecure http service port: 11080
Ports can be changed with environment variables.
https: $SCRYPTED_SECURE_PORT
http : $SCRYPTED_INSECURE_PORT
Certificate can be modified via tls.createSecureContext options in
JSON file located at SCRYPTED_HTTPS_OPTIONS_FILE environment variable:
export SCRYPTED_HTTPS_OPTIONS_FILE=/path/to/options.json
https://nodejs.org/api/tls.html#tlscreatesecurecontextoptions
#######################################################
[19:53:17] INFO: Starting NGinx...
plugin successfully loaded
i Autoupdate Plugins automation is waiting for event trigger.
i Autoupdate Plugins event will fire at Mon May 09 2022 03:15:00 GMT+0200 (Central European Summer Time)
i Scrypted Core loaded @scrypted/core
native dependencies (up to date). jimp
plugin successfully loaded
i Rebroadcast Plugin loaded @scrypted/prebuffer-mixin
i Rebroadcast Plugin Rebroadcaster scheduled for restart at 2AM: 367 minutes
plugin successfully loaded
i Snapshot Plugin loaded @scrypted/snapshot
plugin successfully loaded
i WebRTC Plugin loaded @scrypted/webrtc
python: python3.9
requirements.txt (up to date)
paho-mqtt==1.6.1
requests
Setting plugin verbosity to Normal
i Arlo Camera Plugin loaded @scrypted/arlo
2022-05-08 19:53:21,492 ArloProvider INFO: Trying to initialize Arlo client...
2022-05-08 19:53:21,493 ArloProvider INFO: Initialized Arlo client for arlo@gammelgravlingen.com, reusing stored auth headers
2022-05-08 19:53:21,495 ArloProvider INFO: Discovering devices...
2022-05-08 19:53:21,720 ArloProvider INFO: Discovered 1 basestations
2022-05-08 19:53:21,889 ArloProvider INFO: Discovered 2 cameras
[Ute] prebuffer sessions starting in 5 seconds
[Inne] prebuffer sessions starting in 5 seconds
2022-05-08 19:53:22,390 arlo INFO: Will send heartbeat to the following basestations: ['']
[Ute] stream Cloud RTSP will be rebroadcast on demand.
[Ute] Default Stream: default Cloud RTSP
[Inne] stream Cloud RTSP will be rebroadcast on demand.
[Inne] Default Stream: default Cloud RTSP
[08/May/2022:19:54:21 +0200] 302 GET / HTTP/1.1
rejected request false undefined /endpoint/@scrypted/core/engine.io/api/?EIO=4&transport=polling&t=O2apj_k
[08/May/2022:19:54:22 +0200] 401 GET /endpoint/@scrypted/core/engine.io/api/?EIO=4&transport=polling&t=O2apj_k HTTP/1.1
[08/May/2022:19:54:22 +0200] 200 GET /login HTTP/1.1
[08/May/2022:19:54:24 +0200] 200 GET /endpoint/@scrypted/core/public/service-worker.js HTTP/1.1
[08/May/2022:19:54:24 +0200] 200 GET /endpoint/@scrypted/core/public/precache-manifest.05473cf01ac8c323a805196fc2a338c6.js HTTP/1.1
```

And then the login screen:
![image](https://user-images.githubusercontent.com/21142447/167309013-4c66f64c-1d18-4d6d-a1a1-ccf7c86c8ff7.png)

I'm using patrik/test as login and password here for simplicity. Hit login and:

```
[08/May/2022:19:55:50 +0200] 200 POST /login HTTP/1.1
[08/May/2022:19:55:51 +0200] 200 GET /endpoint/@scrypted/core/engine.io/api/?EIO=4&transport=polling&t=O2aq3dj HTTP/1.1
[08/May/2022:19:55:51 +0200] 200 GET /login HTTP/1.1
[08/May/2022:19:55:51 +0200] 200 GET /endpoint/@scrypted/core/engine.io/api/?EIO=4&transport=polling&t=O2aq3fw&sid=51yNjwjXIBKm79ruAAAA HTTP/1.1
[08/May/2022:19:55:51 +0200] 200 POST /endpoint/@scrypted/core/engine.io/api/?EIO=4&transport=polling&t=O2aq3ir&sid=51yNjwjXIBKm79ruAAAA HTTP/1.1
[08/May/2022:19:55:51 +0200] 200 GET /endpoint/@scrypted/core/engine.io/api/?EIO=4&transport=polling&t=O2aq3it&sid=51yNjwjXIBKm79ruAAAA HTTP/1.1
i Scrypted Core requested listen Ute null
i Scrypted Core requested listen Inne null
2022-05-08 19:55:51,964 ArloCamera[A0949B7WA21EF] INFO: None requested refresh
2022-05-08 19:55:52,122 ArloCamera[A0949B7YABBEF] INFO: None requested refresh
2022-05-08 19:55:52,396 ArloCamera[A0949B7WA21EF] INFO: Taking picture
2022-05-08 19:55:52,400 ArloCamera[A0949B7YABBEF] INFO: Taking picture
[08/May/2022:19:55:53 +0200] 200 GET /endpoint/@scrypted/core/public/service-worker.js HTTP/1.1
[08/May/2022:19:55:53 +0200] 200 GET /endpoint/@scrypted/core/public/precache-manifest.05473cf01ac8c323a805196fc2a338c6.js HTTP/1.1
```

This is what I'm seeing in the network tab of the developer tools:

![image](https://user-images.githubusercontent.com/21142447/167309104-a300d5a5-fffa-4ab1-a946-1f18577d8b03.png)

Some of the logs above are related to the Arlo plugin but you should expect to see the same.

Let's go through your environment and see what we can find. Are you runnings this with a Nabu Casa subscription or an IP-address or DNS-service to access your device from the outside?

**Comment by ggravlingen** [comment]
Did you delete the entire folder? There's a .db-file in there that you won't see in VS-code:

![image](https://user-images.githubusercontent.com/21142447/167309293-ea06e5e0-4769-4f25-a901-45e59bd90c5b.png)

**Comment by newdaddy46052** [comment]
> Did you delete the entire folder? There's a .db-file in there that you won't see in VS-code:
> 
> ![image](https://user-images.githubusercontent.com/21142447/167309293-ea06e5e0-4769-4f25-a901-45e59bd90c5b.png)

So as far as my environment goes im not using nabu or any way to access from outside my LAN. im just getting started with this and im willing to do a fresh install on my homeassistant blue to see if that fixes it. i had frigate on this  but i just un installed it and deleted the folder in vs code. i dont have many things on here as far as intergrations or add ons. 

and yes i deleted the entire folder of scrypted as i dont see it anymore when i use vscode or the file editor.

**Comment by ggravlingen** [comment]
I'm running out of ideas here. 🤔  I had a theory that it didn't work without a proper SSL certificate, but I'm accessing the add-on using my internal IP without any issues: `https://192.168.15.150:8123/f85504c7_ggravlingen_scrypted/dashboard`.

I'm not convinced doing a fresh reinstall will help, but if you want to give it a go I'm all for it just to rule out some weird config.

**Comment by ggravlingen** [comment]
By the way, username and password seem to be case sensitive. Do you get any error when typing in your login/password or does the page just refresh?

![image](https://user-images.githubusercontent.com/21142447/167313906-bdfe408b-680f-4599-a3a9-e75af255c894.png)

**Comment by newdaddy46052** [comment]
> I'm running out of ideas here. 🤔  I had a theory that it didn't work without a proper SSL certificate, but I'm accessing the add-on using my internal IP without any issues: `https://192.168.15.150:8123/f85504c7_ggravlingen_scrypted/dashboard`.
> 
> 
> 
> I'm not convinced doing a fresh reinstall will help, but if you want to give it a go I'm all for it just to rule out some weird config.

Yeah I have no idea what's going on. So before I did an entire wipe of my HAblue I decided to see if there were any other scrypted repos for HA and I found another and that one seems to work fine. I dont know much of what the difference is between that one and yours but I think it's the ingress stuff. Anyways I got that one working and didn't have the log in issues resolved. I'm gonna try adding yours again and see if it suddenly works. I did make sure it was case sensitive every time when entering name and pw.

**Comment by ggravlingen** [comment]
If you don't need ingress, I'd go for the other version of the add on. No need to complicate things. 😊

**Comment by retiep** [comment]
Just to let you know, I have the same problem.

**Comment by Calimerorulez** [comment]
Same problem here.
